ra2ce/analyses/indirect/origin_closest_destination.py

```diff
@@ -648,9 +652,10 @@ def _find_optimal_routes(
                         )
                     )
             else:
-                list_no_path.append((n, ndat[self.od_key]))
+                if ndat[self.od_key] not in node_checked_has_path:
+                    list_no_path.append((n, ndat[self.od_key]))
 
-        return self.update_origins(origins, list_no_path, name_save.format("A"))
+        return self.update_origins(origins, list_no_path, name_save.format("A")), node_checked_has_path
```

Comment:
This is not really required as you are already altering the in-memory values of `node_checked_has_path`.